### DISCUSSION TITLE
licensing concerns

### DISCUSSION THREAD

**Discussion by tmwine** [community/community#31063] [Other Feature Feedback, Questions, & Ideas]
So maybe this was a noob mistake, but I had found code for a balanced binary search tree in GitHub and incorporated it into my project (relatively deeply). There was no license listed, and I thought that would be OK. However, when I decided to post the project to my GitHub account, I realized no license means you can't redistribute the code, *at all*. So I had to spend a day and a half ripping out all the code with the no-license BST, find a substitute with a license on GitHub, re-fit it to work with all the interfaces I had used the binary tree for in the project, and re-test. Ugh.

I did some research and, apparently, though GitHub (understandably) encourages posters of code to choose a license, around a third or more [do not](https://solutionshub.epam.com/blog/post/examining-open-source-license-usage). I'm not sure posters of code (especially those idly posting while taking courses, etc.) realize that no license means basically no redistribution whatsoever, and that renders their code useless for anyone wanting to incorporate it.

So in the interest of further encouraging submitters to license their code (and to notify silly people like me of the consequences of code without licenses), how about every repo comes with a default "no-license" that expressly states, "this code has no license; copyright is held entirely by the owner; you may not redistribute this code in any way shape or form" (etc.). Like instead of a "LICENSE" document in the repo, an expressly-titled "NO LICENSE / NO DISTRIBUTION" document, by default, with all the necessary legalese?

Also, I'm just now realizing that if I add a license to a repo using a template (say an MIT license), but change the template *in any way*, it is no longer searchable as an MIT license. This occurs even just by putting the copyright name/date in a lower than standard position. I understand the cautions around a someone modifying license text in a legally substantial way, but just the copyright line? People may think they've made their repo visible under a given license, but because of some insubstantial change their code is not registered or searchable by GitHub under that license.

**Comment by jfullstackdev** [comment]
you can't also share it since there is no license at the original repo, much more create a license of yours for that copied repo, you run the risk of copyright infringement
